java/client/src/org/openqa/selenium/logging/LoggingHandler.java

```diff
@@ -36,7 +34,7 @@
 public class LoggingHandler extends Handler {
 
   private static final int MAX_RECORDS = 1000;
-  private LinkedList<LogEntry> records = Lists.newLinkedList();
+  private LinkedList<LogEntry> records = new LinkedList<>();
```

Comment:
`private List<LogEntry> records = new ArrayList<>();`

Reply:
@jleyba , it can't be replaced, because `java.util.LinkedList#remove()` (implementation of `java.util.Deque#remove()`) is used at [LoggingHandler.java#L59](https://github.com/SeleniumHQ/selenium/blob/master/java/client/src/org/openqa/selenium/logging/LoggingHandler.java#L59)

Reply:
Then switch to a `ArrayDeque`. `#getRecords()` is only called in [HandlerBasedLocalLogs](https://github.com/SeleniumHQ/selenium/blob/8450067c9b36acedcc026d9aa1f85f9fbcf72329/java/client/src/org/openqa/selenium/logging/HandlerBasedLocalLogs.java#L41). Based on the usage, ti should be safe to change that to return a Collection instead of a List.